Fotoblog.BLL/Services/PhotosService/PhotoService.cs

```diff
@@ -0,0 +1,185 @@
+                image.Mutate(x => x.Resize(width, height, KnownResamplers.Lanczos3));
+
+                using (var outStream = File.Create(path))
+                {                 
+                    await image.SaveAsync(outStream, GetImageEncoder());
+                }
+            }
+        }
+
+        private IImageEncoder GetImageEncoder()
+        {
+            switch(_uploadedFileExtension)
+            {
+                case ".jpg":
+                    return new JpegEncoder();
+                case ".png":
+                    return new PngEncoder();
+                default:
+                    throw new Exception();
+            }
+        }
+
+        private async Task saveOriginalPhoto(string path, IFormFile file)
+        {
+            using (var stream = File.Create(path))
+            {
+                await file.CopyToAsync(stream);
+            }
+        }
+
+        private void doInputValidation(NewPhotoDataVm newPhotoDataVm)
+        {
+            if(newPhotoDataVm.file == null)
+            {
+                throw new ArgumentException();
+            }
+
+            _uploadedFileExtension = Path.GetExtension(newPhotoDataVm.file.FileName).ToLowerInvariant();
+
+            if (!isFileExtensionAllowed() || !isFileContentTypeAllowed(newPhotoDataVm.file.ContentType))
+            {
+                throw new ArgumentException();
+            }
+
+            if (newPhotoDataVm.file.Length == 0)
+            {
+                throw new ArgumentException();
+            }
+
+            if( string.IsNullOrEmpty(newPhotoDataVm.title) ||
+                newPhotoDataVm.title.Length < 2 || newPhotoDataVm.title.Length > 15 )
+            {
+                throw new ArgumentException();
+            }
+            else
+            {
+                newPhotoDataVm.title =  WebUtility.HtmlEncode(newPhotoDataVm.title);
+            }
+
+            if (!string.IsNullOrEmpty(newPhotoDataVm.description) )
+            {
+                if(newPhotoDataVm.description.Length > 30)
+                {
+                    throw new ArgumentException();
+                }
+                else
+                {
+                    newPhotoDataVm.description = WebUtility.HtmlEncode(newPhotoDataVm.description);
+                }
+            }
+        }
+
+        private bool isFileExtensionAllowed()
+        {
+            
+            return (
+                string.IsNullOrEmpty(_uploadedFileExtension)
+                ||!_allowedExtensions.Contains(_uploadedFileExtension)
+                ) ? false : true;
+        }
+
+        private bool isFileContentTypeAllowed(string contentType)
+        {
+
+            return (
+                string.IsNullOrEmpty(contentType)
+                || !_allowedImgContentTypes.Contains(contentType)
+                ) ? false : true;
+        }
+
+        private void PreparePhotoSaveLocation()
+        {
+            do
```

Comment:
Zamiast kręcić pętlą i używać Path.GetRandomFileName() licząc na to, że zwróci nazwę, której jeszcze nie było, ja bym stworzył nowy guid i przypisał go jako nazwę pliku.

Reply:
słusznie, dużo wydajniejsze rozwiązanie, dzięki